File: src/main/java/leetcode/offer/Offer3.java
package leetcode.offer;

import java.util.HashMap;
import java.util.Map;

/**
 * @author jingxinwu
 * @date 2021-11-19 9:06 下午
 */
public class Offer3 {

    public int findRepeatNumber(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
            if (map.containsKey(num)) {
                int count = map.get(num) + 1;
                map.put(num, count);
            } else {
                map.put(num, 1);
            }
        }
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() > 1) {
                return entry.getKey();
            }
        }
        return -1;
    }

}

File: src/main/GO/leetcode/lists/offer/Offer3.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2022/12/27 15:28
 */

func findRepeatNumber(nums []int) int {
	if len(nums) == 0 {
		return -1
	}
	mp := make(map[int]bool)
	for i := 0; i < len(nums); i++ {
		if mp[nums[i]] {
			return nums[i]
		}
		mp[nums[i]] = true
	}
	return -1
}
